src/modules/configs/config.py

```diff
@@ -0,0 +1,74 @@
+from .templates import final
+from modules.global_vars import CONFIG_RULES
+
+
+class MergedConf:
```

Comment:
Test cases required:
- unrecognized key
- missing mandatory key
- missing non-mandatory
- invalid data type
- too long value
- too short value
- rule undefined
- mode undefined
- order undefined
- unexpected mode
- unexpected conf name in order
- exclusive rule Exception, positive, None
- inclusive rule positive, None
- priority rule positive, None

Reply:
Test cases required for MergedConf:
- rule undefined
- mode undefined
- order undefined
- unexpected mode
- unexpected conf name in order
- exclusive rule Exception, positive, None
- inclusive rule positive, None
- priority rule positive, None
- inclusive rule default
- priority rule default

Reply:
- Try to use `@pytest.mark.parametrize` as much as possible.
- Test data should reside in tests/data/(modules)/configs/...
- File/dir structure under test data is up to you - please try to separate them according to some logic (don't throw everything in the same folder) and please try to avoid creating json files as much as possible (some test cases can easily be stored in python files)
- Feel free to extend test cases - for now more is better :)

Reply:
One more comment: feel free to implement a different solution as long as it's efficient and code is clean. This is just a draft to present expected behaviour and results.